### DISCUSSION TITLE
Caching behaviour for ACLs

### DISCUSSION THREAD

**Discussion by on-delete** [emqx/emqx#6413] [Q&A] [answered]
We started implementing a service to handle the ACL requests from the emqx_auth_http plugin and we recognized, that in case a client is publishing a message to a specific topic very often, the ACL request endpoint is triggered also very often. I did some tests and it looked like the frequency was 1 request per minute.

So is there any caching behavior of the emqx, that it remembers for clients the already granted ACLs (e.g. client 'A' is allowed to publish messages to topic 'topic/1')? Is this maybe configurable? In a scenario, where many clients sending many messages in a short period of time, the requests for the ACLs could stress out our service alot. We are using the 4.3.8 version at the moment.

**Comment by zmstone** [comment, marked as answer]
https://docs.emqx.io/en/broker/v4.3/advanced/acl.html#acl-cache

**Reply by on-delete** [reply]
Thanks, missed this documentation.
